docs/CodingStandardPhilosophy.md

```diff
@@ -77,3 +79,100 @@ repeating (visual) patterns should usually be avoided.  This is an aspect
 of the DRY (do not repeat yourself) principle in software engineering,
 since any repetition tends to lead to repeating patterns that are then
 harder to read.
+
+## Commenting the code
+
+Code comments should be meaningful and aid to the understanding of the
+code. We use [Doxygen](http://www.doxygen.nl/manual/docblocks.html)
+style comments. Comments should reflect the programmer's intention
+(rather than enumerating the code) and the reasoning behind certain
+decisions. Comments should also capture invariants that are not
+directly visible in the code. There are arguments that code should be
+self-documenting, and indeed we encourage the writing of clean
+self-explanatory code. Comments should contribute the additional
+meaning that helps extensibility and maintainability by a large group
+of developers.
+
+## Handling errors
```

Comment:
This is too detailed for the coding standards. The longer this section, the less likely it is for anyone to read it.
Also, we should keep this very high level; these are really implementation details and not principles.

Reply:
Where do we put the details?

Reply:
We don't have a good place. The readme, where you removed some details from, isn't great either.
Some of this is also in the compiler-design powerpoint, but you can't find that using search.
People will have to read the code anyway to understand how to report errors, so perhaps this is implied.